source/fundamentals/gridfs.txt

```diff
@@ -4,7 +4,369 @@ GridFS
+
+      if bytes, err = uploadStream.Write(fileContent); err != nil {
+         panic(err)
+      }
+
+      fmt.Printf("New file uploaded with %d bytes written", bytes)
+
+   .. output::
+      :language: none
+      :visible: false
+
+      New file uploaded with 1000 bytes written
+
+.. _golang-retrieve-info:
+
+Retrieve File Information
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can retrieve file metadata stored in the ``files`` collection of the GridFS
+bucket. Each document in the ``files`` collection contains the following
+information:
+
+- The file ID
+- The file length
+- The maximum chunk size
+- The upload date and time
+- The file name
+- A ``metadata`` document in which you can store any other information
+
+To retrieve file data, call the ``Find()`` method on a ``GridFSBucket``
+instance. You can pass a query filter as an argument to ``Find()`` to match
+only certain file documents.
+
+.. note::
+
+   The ``Find()`` method requires a query filter as a parameter. To match all 
+   documents in the ``files`` collection, pass an empty query filter to ``Find()``.
+
+The following example retrieves the file name and length of documents in the
+``files`` collection with ``length`` values greater than ``1500``:
+
+.. code-block:: go
+
+   filter := bson.D {{"length", bson.D{{"$gt", 1500}}}
+   cursor, err := bucket.Find(filter)
+
+   type gridfsFile struct {
+      Name   string `bson:"filename"`
+      Length int64  `bson:"length"`
+   }
+   var foundFiles []gridfsFile
+   if err = cursor.All(context.TODO(), &foundFiles); err != nil {
+      log.Fatal(err)
+   }
+
+   for _, file := range foundFiles {
+      fmt.Printf("filename: %s, length: %d\n", file.Name, file.Length)
+   }
+
+.. _golang-download-files:
+
+Download Files
+~~~~~~~~~~~~~~
+
+You can download a GridFS file in one of the following ways:
+
+- Use the ``DowloadToStream()`` method to download a file to an output stream.
+- Use the ``OpenDownloadStream()`` method to open an input stream.
+
+Download a File to an Output Stream
+```````````````````````````````````
+
+You can download a file in a GridFS bucket directly to an output stream using the
+``DownloadToStream()`` method. ``DownloadToStream()`` takes an ``ObjectID`` value
+and an ``io.Writer`` as parameters. The method downloads the file with the specified
+``ObjectID`` and writes to the ``io.Writer``.
+
+The following example downloads a file with an ``_id`` value of ``62e005408bc`` and
+writes to a file buffer:
+
+.. code-block:: go
+
+   id, err = primitive.ObjectIDFromHex("62e005408bc")
+   fileBuffer := bytes.NewBuffer(nil)
+   if _, err := bucket.DownloadToStream(id, fileBuffer); err != nil {
+      log.Fatal(err)
+   }
+
+Download a File to an Input Stream
+``````````````````````````````````
+
+You can download a file in a GridFS bucket to memory with an input stream using
+the ``OpenDownloadStream()`` method. ``OpenDownloadStream()`` takes a ``ObjectID``
```

Comment:
```suggestion
the ``OpenDownloadStream()`` method. ``OpenDownloadStream()`` takes a file ID
```

The `fileID` argument can be any BSON type.

Reply:
I replaced ObjectId with file ID.